### PR TITLE
Improve reports

### PR DESCRIPTION
**Pull request by cdebourcy**

### CODE REVIEW

File: app/controllers/pipeline_outputs_controller.rb [hunk 1]
```diff
@@ -29,6 +29,8 @@ def create
 
     respond_to do |format|
       if @pipeline_output.save
+        @pipeline_output.name = ['ID#', @pipeline_output.id, ' (', @pipeline_output.sample.name, ')'].join('')
```
Comment (ryanking) [inline app/controllers/pipeline_outputs_controller.rb]: why not do this before the first save?

Comment (ryanking) [reply app/controllers/pipeline_outputs_controller.rb]: note that, as written, if the second save fails we'll not handle the error gracefully

File: app/views/backgrounds/show.html.erb [hunk 2]
```diff
@@ -6,11 +6,18 @@
 </p>
 
 <p>
-  <strong>Used in report(s):</strong>
-  <% @background.reports[0...-1].each do |r| %>
-    <%= link_to r.name, r if r %>, 
+  <% if @background.reports.count == 0 %>
```
Comment (ryanking) [inline app/views/backgrounds/show.html.erb]: You can simplify this whole block to:

```
<strong>Used in <%= pluralize(@background.reports.count, report)%>:</strong>
<%= @background.reports.map do |r|
  link_to(r.name, r)
end.join(',')
%>
```

File: app/views/pipeline_outputs/show.html.erb [hunk 3]
```diff
@@ -26,19 +26,29 @@
 </p>
 
 <p>
-  <strong>Report(s):</strong>
-  <% @pipeline_output.reports[0...-1].each do |r| %>
-    <%= link_to r.name, r if r %>, 
+  <% if @pipeline_output.reports.count == 0 %>
```
Comment (ryanking) [inline app/views/pipeline_outputs/show.html.erb]: Similiar to above, you can simplify this.

File: app/views/projects/show.html.erb [hunk 4]
```diff
@@ -6,11 +6,18 @@
 </p>
 
 <p>
-  <strong>User(s):</strong>
-  <% @project.users[0...-1].each do |u| %>
-    <%= link_to u.name, u if u%>, 
+  <% if @project.users.count == 0 %>
```
Comment (ryanking) [inline app/views/projects/show.html.erb]: simplify

File: db/migrate/20170906201622_create_backgrounds_pipeline_outputs.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class CreateBackgroundsPipelineOutputs < ActiveRecord::Migration[5.1]
+  def change
+    create_table :backgrounds_pipeline_outputs do |t|
+      t.integer :background_id
```
Comment (ryanking) [inline db/migrate/20170906201622_create_backgrounds_pipeline_outputs.rb]: these should be bigint

File: app/controllers/pipeline_outputs_controller.rb (hunk 1 above)
Comment (cdebourcy) [reply app/controllers/pipeline_outputs_controller.rb]: Can't access pipeline_output.id until it's been saved!

File: app/views/backgrounds/show.html.erb (hunk 2 above)
Comment (cdebourcy) [reply app/views/backgrounds/show.html.erb]: cool!

File: app/controllers/pipeline_outputs_controller.rb (hunk 1 above)
Comment (ryanking) [reply app/controllers/pipeline_outputs_controller.rb]: Oh i understand now.

There's no reason to have a `name` column in the db. Just create a method called `#name` in PipelineOutput like so:

```
def name
  return ['ID#', self.id, ' (', self.sample.name, ')'].join('')
end
```

File: db/migrate/20170906201622_create_backgrounds_pipeline_outputs.rb (hunk 5 above)
Comment (cdebourcy) [reply db/migrate/20170906201622_create_backgrounds_pipeline_outputs.rb]: ok

File: app/views/projects/show.html.erb (hunk 4 above)
Comment (cdebourcy) [reply app/views/projects/show.html.erb]: ok

File: app/views/pipeline_outputs/show.html.erb (hunk 3 above)
Comment (cdebourcy) [reply app/views/pipeline_outputs/show.html.erb]: ok

**Pull request merged**